rustls/src/suites.rs

```diff
@@ -214,6 +214,51 @@ pub(crate) fn compatible_sigscheme_for_suites(
         .any(|&suite| suite.usable_for_signature_algorithm(sigalg))
 }
 
+/// Secrets for transmitting/receiving data over a TLS session. These can be
+/// used to configure kTLS for a socket, for example.
+#[cfg(feature = "extract_secrets")]
+pub struct ExtractedSecrets {
+    /// secrets for the "tx" (transmit) direction
+    pub tx: (u64, AlgorithmSecrets),
+
+    /// secrets for the "rx" (receive) direction
+    pub rx: (u64, AlgorithmSecrets),
+}
+
+/// Secrets specific to an AEAD algorithm. These are traffic secrets,
```

Comment:
Same as above. Maybe expand a little on how these might get use (for example, kTLS).